src/tools/rust-analyzer/crates/hir-ty/src/diagnostics/match_check/deconstruct_pat.rs

```diff
@@ -41,6 +41,9 @@
 //! or-patterns; instead we just try the alternatives one-by-one. For details on splitting
 //! wildcards, see [`SplitWildcard`]; for integer ranges, see [`SplitIntRange`].
 
+// This uses uninhabited types to encode statically unused variants.
+#![allow(unreachable_patterns)]
```

Comment:
Drive by comment, but the fact that changing rustc requires atomically changing `rust-analyzer` doesn't feel right to me. Ideally, this change is done by ra devs in ra repo when ra upgrades to the new rust version. 

Perhaps rust-lang/rust build system should build rust-analyzer with `--cap-lints allow`? From rustc point of view, rust-analyzer pretty much is a dependency which might not have upgraded to the latest compiler yet.